libfwupd/fwupd-client.c

```diff
@@ -121,13 +123,123 @@ fwupd_client_curl_helper_free (FwupdCurlHelper *helper)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(FwupdCurlHelper, fwupd_client_curl_helper_free)
 #endif
 
+typedef struct {
+	FwupdClient	*self;
+	gchar		*property_name;
+	guint		 signal_id;
+	FwupdDevice	*device;
+} FwupdClientContextHelper;
+
+static void
+fwupd_client_context_helper_free (FwupdClientContextHelper *helper)
+{
+	g_clear_object (&helper->device);
+	g_object_unref (helper->self);
+	g_free (helper->property_name);
+	g_free (helper);
+}
+
+/* main_ctx == g_main_context_get_thread_default() */
```

Comment:
You could encode that as an assertion, for added safety:
```
g_return_if_fail (g_main_context_is_owner (priv->main_ctx));
```

Might be clearer to phrase the comment as `/* Always executed in the main context given by priv->main_ctx */`

Reply:
Agree, fixed, thanks.